shared/test/loop_mocha_utils.js

```diff
@@ -83,6 +83,12 @@ var LoopMochaUtils = (function(global, _) {
     });
   };
 
+  syncThenable.resolve = function(result) {
```

Comment:
Why doesn't this get overwritten by the "this.resolve" dynamic function assignment inside the constructor.

As a optional nit, it would be nifty to rename syncThenable to SyncThenable, since it's a constructor.